test/entities/swap.test.ts

```diff
@@ -1,126 +1,229 @@
+            outputAmountRaw: 1000000n,
+        };
+        test('should throw if paths are not all same Balancer version', () => {
+            expect(() => {
+                const pathV3: Path = {
+                    ...pathWethDai,
+                    balancerVersion: 3,
+                };
+                new Swap({
+                    chainId: ChainId.MAINNET,
+                    paths: [pathWethDai, pathV3],
+                    swapKind: SwapKind.GivenIn,
+                });
+            }).toThrowError(
+                'Unsupported swap: all paths must use same Balancer version.',
+            );
+        });
+        describe('path inputs show all start/end with same token', () => {
+            test('should throw if paths start with different token', () => {
+                expect(() => {
+                    const pathUsdcDai: Path = {
+                        ...pathWethDai,
+                        tokens: [tokens[2], tokens[1]],
+                    };
+                    new Swap({
+                        chainId: ChainId.MAINNET,
+                        paths: [pathWethDai, pathUsdcDai],
+                        swapKind: SwapKind.GivenIn,
+                    });
+                }).toThrowError(
+                    'Unsupported swap: all paths must start/end with same token.',
+                );
+            });
+            test('should throw if paths end with different token', () => {
+                expect(() => {
+                    const pathWethUsdc: Path = {
+                        ...pathWethDai,
+                        tokens: [tokens[0], tokens[2]],
+                    };
+                    new Swap({
+                        chainId: ChainId.MAINNET,
+                        paths: [pathWethDai, pathWethUsdc],
+                        swapKind: SwapKind.GivenIn,
+                    });
+                }).toThrowError(
+                    'Unsupported swap: all paths must start/end with same token.',
+                );
             });
-            const amountIn = swap.inputAmount; // In production code, this would be the expected amountIn returned from the query
-            const slippage = Slippage.fromPercentage('1');
-            const limits = swap.limits(slippage, amountIn);
-            const expected = amountIn.mulDownFixed(
-                parseUnits(`${1 + slippage.decimal}`, 18),
-            ).amount;
+        });
+    });
+
+    describe('buildCall max/min amounts', () => {
+        describe('GivenIn', () => {
+            test('18decimals>6decimals: minAmountOut to be 0.1% less then expected', () => {
+                const swap = new Swap({
+                    chainId: ChainId.MAINNET,
+                    paths: [pathTo6Decimals],
+                    swapKind: SwapKind.GivenIn,
+                });
+                const slippage = Slippage.fromPercentage('0.1');
+                const tokenOut = new Token(
+                    1,
```

Comment:
To avoid using incorrect ids, we we could create a `chainId` variable with a broader scope and reuse. What do you think?